File: backend/app/schemas/payment.py
import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class PaymentCreate(BaseModel):
    user_id: uuid.UUID
    amount: Decimal = Field(gt=0, max_digits=10, decimal_places=2)
    date: date
    note: str | None = Field(default=None, max_length=255)


class PaymentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    amount: Decimal
    date: date
    note: str | None
    recorded_by: uuid.UUID
    created_at: datetime


class BalanceResponse(BaseModel):
    user_id: uuid.UUID
    total_paid: Decimal
    total_billed: Decimal
    balance: Decimal

File: frontend/lib/core/models/payment.dart
class Payment {
  final String id;
  final String userId;
  final double amount;
  final DateTime date;
  final String? note;
  final String recordedBy;
  final DateTime createdAt;

  Payment({
    required this.id,
    required this.userId,
    required this.amount,
    required this.date,
    this.note,
    required this.recordedBy,
    required this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id'] as String,
        userId: json['user_id'] as String,
        amount: double.parse(json['amount'].toString()),
        date: DateTime.parse(json['date'] as String),
        note: json['note'] as String?,
        recordedBy: json['recorded_by'] as String,
        createdAt: DateTime.parse(json['created_at'] as String),
      );
}

class Balance {
  final String userId;
  final double totalPaid;
  final double totalBilled;
  final double balance;

  Balance({
    required this.userId,
    required this.totalPaid,
    required this.totalBilled,
    required this.balance,
  });

  factory Balance.fromJson(Map<String, dynamic> json) => Balance(
        userId: json['user_id'] as String,
        totalPaid: double.parse(json['total_paid'].toString()),
        totalBilled: double.parse(json['total_billed'].toString()),
        balance: double.parse(json['balance'].toString()),
      );
}
